### PR TITLE
Misc. Documentation Improvements & Changes With Focus on Selectors & Tests

### PR DESCRIPTION
**Pull request by jpjsonic**
Pretty big improvements to our documentation. Touched on a lot of files, but the main things are:

1. Overhauled the "getting started landing page" including a video walking through init.
2. Added interactivity topic
3. Added testing topic

### CODE REVIEW

File: mkdocs/topics/testing.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# Testing
```
Comment (jpjsonic) [inline mkdocs/topics/testing.md:1]: @tgsoverly here's some aspiration docs for tests. I used syntax for the trace references that was along the lines of what we discussed yesterday.

File: mkdocs/topics/targets.md [hunk 2]
```diff
@@ -1,20 +1,36 @@
...
 
-An example of a target might be the database that your application runs on.  We recommend that you think of a target in your project by the what it is used for and not by its environment.  
+This is really useful for joining together data that lives in different sources. 
 
-For example lets say you have an application that stores its data in a PostgreSQL database call `app_db` and runs locally, in staging, and in production. This means you have three separate databases, but they all used to run your application, so that is a single `target`.  The following might be how you set that up in your project:
+Once your targets are set up, you can bring data together in a single chart with traces that originate from different targets, or through sqlite queries that leverage tables from multiple targets (see [LocalMergeModel](/../reference/configuration/Models/LocalMergeModel/)), or through writing csvs to stdout (see [CsvScriptModels](/../reference/configuration/Models/CsvScriptModel/)).
+
+
+## Environments 
+We recommend that you think of a target in your project as something that represents a type of data source and _not_ its a type of data source + it's environment.
```
[the remark this answers is not in the document]
Comment (tgsoverly) [reply mkdocs/topics/targets.md:16]: "We recommend that you think of a `target` as a consistent data source you have across environments."

File: mkdocs/topics/testing.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+The combination of python based boolean statements and access to all trace data makes it _super fast_ all while enabling complex assertions about your data. 
+
+!!! example
+
+    === "Revenue Sums Match"
+   
+        ``` yaml
+        tests:
+          - name: revenue-date-grains-match
+            logic: "numpy.sum( $(project.traces['revenue-per-week'].props.y) ) = numpy.sum( $(project.traces['revenue-per-month'].props.y) )"
```
Comment (tgsoverly) [inline mkdocs/topics/testing.md:19]: Does this work with cohorts?  Since there are multiple "props.y"

File: visivo/commands/init_phase.py [hunk 4]
```diff
@@ -89,7 +89,6 @@ def init_phase():
         )
 
     if target_type == snowflake_type:
-        host = click.prompt("? Database host", type=str)
```
Comment (tgsoverly) [inline visivo/commands/init_phase.py:92]: Why is this gone?

Comment (jpjsonic) [reply visivo/commands/init_phase.py:92]: There's no host attribute in a snowflake connection. The `account` serves a similar purpose.

File: mkdocs/topics/testing.md (hunk 3 above)
Comment (jpjsonic) [reply mkdocs/topics/testing.md:19]: We'll probably need to re-write this testing docs page once we finalize our syntax for context. I don't think people would need to specify a cohort though, we can flatten out the trace data and make the cohort a column again.

**Pull request merged**